static/js/topic_page/map_tile.tsx

```diff
@@ -121,141 +112,143 @@ export function MapTile(props: MapTilePropType): JSX.Element {
+    isPerCapita: boolean,
+    chartTitle: string,
+    setChartData: (data: ChartData) => void,
+    scaling?: number
+  ): void {
+    const dataValues = {};
+    const metadata = {};
+    const sources: Set<string> = new Set();
+    const dates: Set<string> = new Set();
+    for (const geoFeature of rawData.geoJson.features) {
+      const placeDcid = geoFeature.properties.geoDcid;
+      const placeChartData = getPlaceChartData(
+        rawData.placeStat,
+        placeDcid,
+        isPerCapita,
+        rawData.population,
+        rawData.metadataMap
       );
+      if (_.isEmpty(placeChartData)) {
+        continue;
+      }
+      let value = placeChartData.value;
+      if (scaling !== null && scaling !== undefined) {
+        value = value * scaling;
+      }
+      dataValues[placeDcid] = value;
+      metadata[placeDcid] = placeChartData.metadata;
+      dates.add(placeChartData.date);
+      placeChartData.sources.forEach((source) => {
+        if (!_.isEmpty(source)) {
+          sources.add(source);
+        }
+      });
     }
-    return place.name + ": " + value;
-  };
-  drawChoropleth(
-    props.tileId,
-    chartData.geoJson,
-    CHART_HEIGHT,
-    width,
-    chartData.dataValues,
-    props.unit,
-    colorScale,
-    _.noop,
-    getTooltipHtml,
-    () => false,
-    true,
-    true,
-    props.isUsaPlace,
-    props.placeDcid
-  );
-}
+    setChartData({
+      dataValues,
+      metadata,
+      sources,
+      chartTitle: getTitle(Array.from(dates), chartTitle, false),
+      geoJson: rawData.geoJson,
+    });
+  }
 
-function getSourcesJsx(sources: Set<string>): JSX.Element[] {
-  const sourceList: string[] = Array.from(sources);
-  const seenSourceDomains = new Set();
-  const sourcesJsx = sourceList.map((source, index) => {
-    const domain = urlToDomain(source);
-    if (seenSourceDomains.has(domain)) {
-      return null;
-    }
-    seenSourceDomains.add(domain);
-    return (
-      <span key={source}>
-        {index > 0 ? ", " : ""}
-        <a href={source}>{domain}</a>
-      </span>
+  function drawMap(chartData: ChartData): void {
+    const mainStatVar = props.statVarMetadata.statVars[0].main;
+    const width = SVG_CONTAINER_ELEMENT.current.offsetWidth;
+    const colorScale = getColorScale(mainStatVar, chartData.dataValues);
+    const getTooltipHtml = (place: NamedPlace) => {
+      let value = "Data Missing";
+      if (place.dcid in chartData.dataValues) {
+        value = formatNumber(
+          Math.round(
+            (chartData.dataValues[place.dcid] + Number.EPSILON) * 100
+          ) / 100,
+          props.statVarMetadata.unit
+        );
+      }
+      return place.name + ": " + value;
+    };
+    drawChoropleth(
+      props.id,
+      chartData.geoJson,
+      CHART_HEIGHT,
+      width,
+      chartData.dataValues,
+      props.statVarMetadata.unit,
+      colorScale,
+      _.noop,
+      getTooltipHtml,
+      () => false,
+      true,
+      true,
+      true,
```

Comment:
we will need to pass in the bool isUsaPlace, since we'll want these pages for non-us places (we will have for india as a start)

Reply:
Reverted